tests/playwright-test/test-serial.spec.ts

```diff
@@ -140,6 +140,40 @@ test('test.describe.serial should work with retry', async ({ runInlineTest }) =>
   ]);
 });
 
+test('test.describe.serial should work with retry when it fails in different places', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    'a.test.ts': `
+      import { test, expect } from '@playwright/test';
+      test.describe.serial('serial suite', () => {
+        test('test1', async ({}, testInfo) => {
+          console.log('\\n%%test1');
+          expect(testInfo.retry).toEqual(0)
+        });
+        test('test2', async ({}, testInfo) => {
+          console.log('\\n%%test2');
+          expect(testInfo.retry).toEqual(1)
+        });
+        test('test3', async ({}) => {
+          console.log('\\n%%test3');
+        });
+        test('test4', async ({}) => {
+          console.log('\\n%%test4');
+        });
+      });
+    `,
+  }, { retries: 1 });
+  expect(result.exitCode).toBe(1);
+  expect(result.passed).toBe(0);
+  expect(result.flaky).toBe(1);
```

Comment:
I believe in this scenario at least two tests `test1` and `test2` are flaky, aren't they?

Reply:
My apologies, read the bug again, this is an intentional change. I'll discuss in the team meeting if this is change that we want.

Reply:
Discussed this in the meeting, we'll consider all the implications and other possibilities and @dgozman will come back to you.